gson/src/org/immutables/gson/stream/JsonGeneratorWriter.java

```diff
@@ -145,7 +150,25 @@ public JsonWriter value(Number value) throws IOException {
         throw new IllegalArgumentException("JSON forbids NaN and infinities: " + value);
       }
     }
-    generator.writeNumber(d);
+    if (value instanceof Integer) {
+      generator.writeNumber((Integer) value);
```

Comment:
you can use `value.intValue()` here.
The implementation of this method will return `this` (for Integer)